### DISCUSSION TITLE
AWS Stack best practices

### DISCUSSION THREAD

**Discussion by redbugz** [sorry-cypress/sorry-cypress#821] [General]
We are using the AWS Stack as listed here:
https://docs.sorry-cypress.dev/cloud-setup/aws

We have it setup and running, but have a few questions about the best way to manage this.
We are a bit new to some of the AWS technologies in use, so just making sure we understand how to manage this well.

First question is it appears the Mongo instance is just hosted locally on the Fargate task instance machine. The docs say that you get 20GB of ephemeral storage for Fargate, the 20GB seems plenty, as that should provide plenty of space, at least 15GB after all the images and data for the other instances. However I can find no docs on what the conditions of "ephemeral" are for Fargate, but my best guess would be each time we would update the sorry-cypress version or redeploy the Fargate task, I'm guessing that storage is lost and all the history would be gone? If so, would it be advisable to attach and EFS volume to the Fargate task, would Mongo do okay using a volume like that, or would it better to switch to MongoDB Atlas or DocumentDB as an alternative, similar to what is being proposed in https://github.com/sorry-cypress/sorry-cypress/pull/812?

Also, by default the URL is http instead of https. We run all of our tools/services on https, so this feels a bit disconcerting or unusual for us. It seems like perhaps this isn't likely to cause any real issues as I don't thing our tests expose any sensitive information, but just curious if it would be nice to provide instructions or parameters to attach a certificate and run this over https instead?

**Comment by agoldis** [comment]
> re: First question is it appears the Mongo instance is just hosted locally 

The stack is compatible with AWS DocumentDB - you can spin an instance and change the configuration  + deploy ECS. See https://github.com/sorry-cypress/sorry-cypress/pull/812

> re:  I'm guessing that storage is lost and all the history would be gone

That's correct, unless you store the data on an external DB / permanent storage

> re:  but just curious if it would be nice to provide instructions or parameters to attach a certificate and run this over https instead?

it would be nice and a great addition to https://github.com/sorry-cypress/sorry-cypress/pull/812
